src/custom/components/Popups/FollowPendingTxPopup/index.tsx

```diff
@@ -9,38 +9,95 @@ import {
   handleFollowPendingTxPopupAtom,
   handleHidePopupPermanentlyAtom,
   showFollowTxPopupAtom,
+  followPendingTxPopupAtom,
+  selectAtom,
+  handleCloseOrderPopupAtom,
 } from 'state/application/atoms'
 import FollowPendingTxPopupUI from './FollowPendingTxPopupUI'
 import { isPending } from 'components/Web3Status'
+import { OrderID } from '@src/custom/api/gnosisProtocol'
 
-function isAnOrder(element: TransactionAndOrder): element is Order & { addedTime: number } {
+interface AddedOrder extends Order {
+  addedTime: number
+}
+
+function isAnOrder(element: TransactionAndOrder): element is AddedOrder {
   return 'inputToken' in element && 'outputToken' in element
 }
 
-const FollowPendingTxPopup: React.FC = ({ children }): JSX.Element => {
+export const useLastPendingOrder = (): { lastPendingOrder: AddedOrder | null; onClose: () => void } => {
   const setShowFollowPendingTxPopup = useUpdateAtom(handleFollowPendingTxPopupAtom)
-  const setHidePendingTxPopupPermanently = useUpdateAtom(handleHidePopupPermanentlyAtom)
-  const showFollowPendingTxPopup = useAtomValue(showFollowTxPopupAtom)
-  const isExpertMode = useIsExpertMode()
+  const setLastOrderClosed = useUpdateAtom(handleCloseOrderPopupAtom)
   const allRecentActivity = useRecentActivity()
-  const isTherePendingOrder = useMemo(() => {
-    if (!isExpertMode) return
+  const lastPendingOrder = useMemo(() => {
     const pendings = allRecentActivity.filter(isPending)
+    const lastOrder = pendings[pendings.length - 1]
+
+    if (!lastOrder || !isAnOrder(lastOrder)) return null
+
+    return lastOrder
+  }, [allRecentActivity])
```

Comment:
I would think something like this exists already... unless it depends on the format exported by `useRecentActivity`
In that case, might be worth to create a hook for a subset of recent activity, like `useRecentActivityOrders` or more specific `useRecentActivityPendingOrders`.

Reply:
I have created [`useRecentActivityLastPendingOrder`](https://github.com/cowprotocol/cowswap/blob/872-hiden-popup-after-refresh/src/custom/hooks/useRecentActivity.ts#L271) to manage this.